invenio_files_rest/contrib/s3.py

```diff
@@ -0,0 +1,112 @@
+# coding=utf-8
+
+from __future__ import absolute_import, print_function
+
+import hashlib
+import tempfile
+from os.path import join
+
+from boto.s3.connection import S3Connection
+from flask import current_app, redirect
+
+from boto.s3.key import Key
+
+from invenio_files_rest.errors import StorageError
+from invenio_files_rest.storage import Storage
+
+
+class AmazonS3(Storage):
+    """
+    Represents a S3 storage for a given object.
+
+    :param object: the object linked to the file
+    :param fileinstance:the file instance.
+    :raise StorageError: if it is missing critical config variable
+    """
+
+    def __init__(self, object, fileinstance):
+        super(AmazonS3, self).__init__(object, fileinstance)
+        if "FILES_REST_AWS_KEY" not in current_app.config:
+            raise StorageError("Missing AWS Key")
+        if "FILES_REST_AWS_SECRET" not in current_app.config:
+            raise StorageError("Missing AWS Secret")
+        self.aws_connection = S3Connection(current_app.config.get(
+            "FILES_REST_AWS_KEY"), current_app.config.get(
+            "FILES_REST_AWS_SECRET"))
+
+    def get_size(self):
+        """
+        Get the size of the file linked to the storage.
+
+        :return: Integer representing the size.
+        """
+        bucket_name, key_name = self.obj.file.uri.replace("s3:", "").split(":")
+        bucket = self.aws_connection.get_bucket(bucket_name)
+        return bucket.lookup(key_name).size
+
+    def open(self):
+        """
+        Return a file descriptor that can be used later.
+
+        :return: file descriptor that can be used later.
+        """
+        temp = tempfile.NamedTemporaryFile()
```

Comment:
Honestly open, was in your first implementation of storage and is not there anymore so I can perhaps delete it.

But I was thinking to it for case where you need local processing, OCR or else. I will seek if it is possible to get a file like object.